application_sdk/observability/observability.py

```diff
@@ -422,38 +418,33 @@ async def _flush_records(self, records: List[Dict[str, Any]]):
                     elif part.startswith("day="):
                         new_df["day"] = int(part.split("=")[1])
 
-                # Merge with existing data if any
-                if existing_df is not None:
-                    df = pd.concat([existing_df, new_df], ignore_index=True)
-                else:
-                    df = new_df
-
-                # Sort by timestamp to maintain order
-                df = df.sort_values("timestamp")
+                # Use new data directly - let ParquetOutput handle consolidation and merging
+                df = new_df
 
-                # Write to parquet file
-                df.to_parquet(
-                    parquet_path,
-                    compression="snappy",
-                    index=False,
-                )
-
-                # Upload to object store
-                with open(parquet_path, "rb") as f:
-                    file_content = f.read()
-                    relative_path = os.path.relpath(parquet_path, TEMPORARY_PATH)
-                    metadata = {
-                        "key": relative_path,
-                        "blobName": relative_path,
-                        "fileName": relative_path,
-                    }
-                    with DaprClient() as client:
-                        client.invoke_binding(
-                            binding_name=DEPLOYMENT_OBJECT_STORE_NAME,
-                            operation=DAPR_BINDING_OPERATION_CREATE,
-                            data=file_content,
-                            binding_metadata=metadata,
-                        )
+                # Use ParquetOutput abstraction for efficient writing and uploading
+                # Set the output path for this partition
+                try:
+                    # Lazy import and instantiation of ParquetOutput
+                    from application_sdk.outputs.parquet import ParquetOutput
+                    parquet_output = ParquetOutput(
+                        output_path=partition_path,  # Set output path during initialization
+                        use_consolidation=True,
+                        retain_local_copy=True  # Keep local files since we're doing multiple operations
```

Comment:
why do we need to retain local temporary files? the operation by default uploads to both buckets (deployment and upstream)